### PR TITLE
Update long haul deployment to use TRC

### PR DESCRIPTION
**Pull request by and-rewsmith**
We are now switching to use the TRC in longhaul. Since the connectivity test script contained logic relevant to the TRC, I discussed with Bear and David and we thought it was appropriate to move the logic for longhaul into the connectivity script and rename it. 

 This PR gates the use of the TRC behind a feature flag because there are TRC code changes necessary for reporting to run correctly. Once we make the necessary changes to the testdashboard and the TRC we can remove the feature flag and remove the longhaul logic from runE2ETest.sh.

### CODE REVIEW

File: e2e_deployment_files/connectivity_deployment.template.json [hunk 1]
```diff
@@ -513,6 +531,12 @@
               },
               "TransportType": {
                 "value": "Mqtt"
+              },
+              "TwinUpdateSize": {
+                "value": "<TwinUpdateSize>"
+              },
+              "TwinUpdateFrequency": {
+                "value": "<TwinUpdateFrequency>"
```
Comment (and-rewsmith) [inline e2e_deployment_files/connectivity_deployment.template.json:539]: I added these params to connectivity as we might want to configure them in the future.

File: scripts/linux/runE2ETest.sh [hunk 2]
```diff
@@ -506,7 +506,6 @@ function process_args() {
 }
 
 function get_hash() {
-    # TODO: testHelper.sh needs to be shared across build pipelines
```
Comment (and-rewsmith) [inline scripts/linux/runE2ETest.sh:509]: This script doesn't exist so I removed all references to it

File: scripts/linux/trcE2ETest.sh [hunk 3]
```diff
@@ -77,18 +86,25 @@ function get_artifact_file() {
 
     local filter
     case "$fileType" in
-        'aziot_edge' ) filter='aziot-edge_*.deb';;
-        'aziot_is' ) filter='aziot-identity-service_*.deb';;
+        'aziot_edge' ) filter='*/aziot-edge_*.deb';;
+        'aziot_is' ) filter='*/aziot-identity-service_*.deb';;
```
Comment (and-rewsmith) [inline scripts/linux/trcE2ETest.sh:90]: This was a bug from the iiot merge

**Review by dylanbronson — APPROVED** [review]
This looks good to me as long as it deploys correctly!

**Comment by and-rewsmith** [comment]
Discussed with Dylan the current status. Deploys everything but there is a problem with name resolution in the docker network, Since this is behind a feature flag I think we can merge now and fix in a separate PR.

**Pull request merged**